### DISCUSSION TITLE
Multiple Control Plane

### DISCUSSION THREAD

**Discussion by Nyuuk** [siderolabs/talos#9055] [Q&A] [answered]
hello, please help me to create a cluster with multiple control planes, I don't understand the contents of the existing documentation. does each controlplane create a new yaml `controlplane` or use only 1 `controlplane.yaml` file. and what about the worker? 
![image](https://github.com/user-attachments/assets/ee1d6daf-dcde-46d5-bdf9-26f3b3245919)

**Comment by smira** [comment, marked as answer]
You take the `controlplane.yaml` which is a base controlplane configuration, and optionally customize it for each of your machines (e.g. you want a custom hostname for each machine), and then apply it.

If your machines don't need machine-specific configuration, you can apply same `controlplane.yaml` to all of them.

Make sure you bootstrap (`talosctl bootstrap`) exactly one of them.

All the steps are available in [Getting Started guide](https://www.talos.dev/v1.7/introduction/getting-started/).

**Reply by Nyuuk** [reply]
so, when `controlplane.yaml` is not edited, control plane will use the same `cluster.controlplane.endpoint` ? how about when i want to configure `vips` ?

**Reply by Nyuuk** [reply]
i have been tried to apply one `controlplane.yaml`. but when i tried to restart 1 of control plane, log of dashboard show ` [talos] task startAllServices (1/1): service "ext-qemu-guest-agent" to be "up"`
![image](https://github.com/user-attachments/assets/62c85076-2a4f-4184-bf37-663850474cae)

i am using Proxmox to learn

**Reply by smira** [reply]
> so, when controlplane.yaml is not edited, control plane will use the same cluster.controlplane.endpoint ? how about when i want to configure vips ?

Cluster endpoint is a cluster-wide thing, it's not specific to the controlplane node. If use the VIP, put it there.

The qemu-guest-agent is unrelated I guess.
